cradle-cassandra/src/main/java/com/exactpro/cradle/cassandra/dao/testevents/PageScopesIteratorProvider.java

```diff
@@ -0,0 +1,79 @@
+package com.exactpro.cradle.cassandra.dao.testevents;
+
+import com.datastax.oss.driver.api.core.cql.BoundStatementBuilder;
+import com.exactpro.cradle.BookInfo;
+import com.exactpro.cradle.cassandra.dao.CassandraOperators;
+import com.exactpro.cradle.cassandra.dao.testevents.converters.PageScopeEntityConverter;
+import com.exactpro.cradle.cassandra.iterators.ConvertingPagedIterator;
+import com.exactpro.cradle.cassandra.resultset.IteratorProvider;
+import com.exactpro.cradle.cassandra.retries.SelectQueryExecutor;
+import com.exactpro.cradle.counters.Interval;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class PageScopesIteratorProvider extends IteratorProvider<String> {
+
+
+    private final CassandraOperators operators;
+    private final ExecutorService composingService;
+    private final SelectQueryExecutor selectQueryExecutor;
+    private final BookInfo book;
+    private final Queue<String> pages;
+    private final Function<BoundStatementBuilder, BoundStatementBuilder> readAttrs;
+
+    public PageScopesIteratorProvider(String requestInfo,
+                                      CassandraOperators operators,
+                                      BookInfo book,
+                                      Interval interval,
+                                      ExecutorService composingService,
+                                      SelectQueryExecutor selectQueryExecutor,
+                                      Function<BoundStatementBuilder, BoundStatementBuilder> readAttrs) {
+        super(requestInfo);
+        this.operators = operators;
+        this.book = book;
+        this.pages = getPagesInInterval(book, interval);
+        this.composingService = composingService;
+        this.selectQueryExecutor = selectQueryExecutor;
+        this.readAttrs = readAttrs;
+    }
+
+    private Queue<String> getPagesInInterval(BookInfo book, Interval interval) {
+        Instant start = interval.getStart();
+        Instant end = interval.getEnd();
+
+        return book.getPages()
+                .stream()
+                .filter(page -> (page.getStarted().isAfter(start) && page.getStarted().isBefore(end))
+                        || (page.getEnded().isAfter(start) && page.getEnded().isBefore(end)))
```

Comment:
page.getEnded() >= start && page.getStarted() <= end

Reply:
It is Java, it doesn't work this way here)

Reply:
Yes) but I offered simple approach to check which can be implemented via compare function